### PR TITLE
feat(testing): Use temporal.download for downloading test server

### LINKED ISSUE #782: [Feature Request] Put test server behind CDN to avoid `rate limit exceeded`
### Is your feature request related to a problem? Please describe.

Users using testing package may get:

`HTTPError: Response code 403 (rate limit exceeded)`

from: 

`node ./scripts/download-test-server.mjs`

https://temporalio.slack.com/archives/C01DKSMU94L/p1658887447564889

### Describe the solution you'd like

Currently downloading straight from GH. Instead use CDN.

### PR DESCRIPTION
**Pull request by bergundy**
- Closes #782
- Add rust-cache action
- Add `WORKFLOW_ID_REUSE_POLICY_TERMINATE_IF_RUNNING`

Also printing worker status in load tests to help debug CI issues.

### CODE REVIEW

File: packages/testing/scripts/download-test-server.mjs [hunk 1]
```diff
@@ -2,10 +2,30 @@ import stream from 'node:stream';
 import util from 'node:util';
```
Comment (lorensr) [inline packages/testing/scripts/download-test-server.mjs:2]: tested manually?

**Review by lorensr — APPROVED** [review]

File: packages/testing/scripts/download-test-server.mjs (hunk 1 above)
Comment (bergundy) [reply packages/testing/scripts/download-test-server.mjs:2]: I tested manually but this path is exercised in ci too

File: packages/testing/scripts/download-test-server.mjs [hunk 2]
```diff
@@ -20,64 +40,28 @@ try {
...
-  }
-  throw new Error(`No prebuilt test server for ${systemPlatform}-${systemArch}`);
-}
+  searchParams: { 'sdk-name': 'typescript', 'sdk-version': pkg.version },
+};
 
-const asset = findTestServerAsset(latestReleaseRes.assets);
-console.log('Downloading test server', { asset: asset.name, outputPath });
+// TODO: use "real" arch when we get arm builds for test server
+const assetUrl = `https://temporal.download/temporal-test-server/temporal-test-server_default_${systemPlatform}_amd64.${assetExt}`;
```
Comment (cretz) [inline packages/testing/scripts/download-test-server.mjs:51]: How often does this run? Only on upgrade? Can this overwrite files? I assume we don't care about downloading something we already have?

**Comment by bergundy** [comment]
For some reason `checkExtends` needed to be bidirectional in this codebase and not in the [TS playground](https://www.typescriptlang.org/play?#code/KYOwrgtgBAglDeBYAUFNUCGUC8UCMANCulAEY5QBMRyAviiqJFAEILHpa6Er3IoAzMCADGAFwCWAexBQRAC2AiA1gFEAHmNAATAM4AeAPoB5AE4SA5gSiGAwlIAOATyjBNO3TbOWAfAAoASnY6BmQFJTV3ED19GGsWfwCAbiA)

**Pull request merged**